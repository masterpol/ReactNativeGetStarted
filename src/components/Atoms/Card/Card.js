import React from 'react'
import styled from '@emotion/native'

const Container = styled.View`
  flex: 1;
  border-radius: 10px;
  padding: 10px;
  box-shadow: -5px 5px 0 ${({ theme }) => theme.colors.gray};
  border: 1px solid ${({ theme }) => theme.colors.gray};
  align-items: center;
  background-color: ${({ theme }) => theme.colors.white};
`

const Card = ({ children, ...props }) => {
  return <Container {...props}>{children}</Container>
}

export default Card
