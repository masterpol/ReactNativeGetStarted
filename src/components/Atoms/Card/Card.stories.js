import React from 'react'
import { storiesOf } from '@storybook/react-native'
import styled from '@emotion/native'
import Theme from '_themes'
import Card from './Card'

const Container = styled.View`
  flex: 1;
  padding: 10px;
`
const Text = styled.Text``

storiesOf('Card', module).add('default', () => (
  <Theme>
    <Container>
      <Card>
        <Text>Some pretty Card</Text>
      </Card>
    </Container>
  </Theme>
))
