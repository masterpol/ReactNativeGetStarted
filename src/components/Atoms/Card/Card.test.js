import React from 'react'
import Theme from '_themes'
import Card from './Card'

describe('Card', () => {
  let props

  beforeEach(() => {
    props = {}
  })

  describe('structure', () => {
    it('should match the snapshot for Card', () => {
      const wrapper = render(
        <Theme>
          <Card testID="card-test" {...props} />
        </Theme>,
      )
      expect(wrapper).toMatchSnapshot()
    })
  })
})
