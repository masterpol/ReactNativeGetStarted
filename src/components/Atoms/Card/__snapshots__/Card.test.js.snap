// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card structure should match the snapshot for Card 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f5f5f5",
        "borderBottomLeftRadius": 10,
        "borderBottomRightRadius": 10,
        "borderColor": "#9e9e9e",
        "borderStyle": "solid",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "borderWidth": 1,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "shadowColor": "#9e9e9e",
        "shadowOffset": Object {
          "height": 5,
          "width": -5,
        },
        "shadowOpacity": 1,
        "shadowRadius": 0,
      },
      undefined,
    ]
  }
  testID="card-test"
/>
`;
